api/test/integration/tavern_utils.py

```diff
@@ -88,3 +88,49 @@ def test_validate_upgrade_result(response, upgraded):
 
 def test_validate_update_latest_version(response):
     assert response.json().get('code', None) == 1749 or response.json().get('code', None) == 1718
+
+
+def test_count_elements(response, n_expected_items):
+    """
+    :param response: Request response
+    :param n_expected_items: Expected number of elements in affeted_items
+    """
+    assert len(response.json()['data']['affected_items']) == n_expected_items
+
+
+def test_expected_value(response, key, expected_values):
+    """
+    :param response: Request response
+    :param key: Key whose value to compare.
+    :param expected_values: Values to be found inside response.
+    """
+    expected_values = set(expected_values.split(',')) if not isinstance(expected_values, list) else set(expected_values)
+
+    for item in response.json()['data']['affected_items']:
+        response_set = set(item[key])
+        assert bool(expected_values.intersection(response_set)), \
+            f'Expected values {expected_values} not found in {item[key]}'
+
+
+def test_mitre_sort(response, attack_id, order):
+    """
+    :param response: Request response
+    :param id: ID to compare.
+    :param order: Values to be found inside response.
+    """
+    if order == 'desc':
+        assert response.json()['data']['affected_items'][0]['id'] > attack_id
+    else:
+        assert response.json()['data']['affected_items'][0]['id'] < attack_id
+
+
+def test_mitre_select(response, select_keys):
```

Comment:
Modify test_select_key_affected_items to work with a single select keys or a list of keys. That way we keep it generic and available for all tests.